pymatgen/io/lobster/inputs.py

```diff
@@ -116,176 +117,145 @@ class Lobsterin(UserDict, MSONable):
+        Raises:
+            KeyError: if keyword is not available.
+        """
+        key = key.strip().lower()
 
-        key_is_unknown = normalized_key.lower() not in map(str.lower, Lobsterin.AVAILABLE_KEYWORDS)
-        if key_is_unknown or normalized_key not in self.data:
-            raise KeyError(f"{key=} is not available")
+        if key not in type(self).AVAILABLE_KEYWORDS:
+            raise KeyError(f"Key {key} is currently not available")
 
-        return self.data[normalized_key]
+        super().__setitem__(key, val.strip() if isinstance(val, str) else val)
 
-    def __contains__(self, key) -> bool:
-        """Implements getitem from dict to avoid problems with different key casing."""
-        normalized_key = next((k for k in self if key.strip().lower() == k.lower()), key)
+    def __getitem__(self, key: str) -> Any:
+        """To avoid cases sensitivity problems."""
+        try:
+            return super().__getitem__(key.strip().lower())
 
-        key_is_unknown = normalized_key.lower() not in map(str.lower, Lobsterin.AVAILABLE_KEYWORDS)
-        return not key_is_unknown and normalized_key in self.data
+        except KeyError as exc:
+            raise KeyError(f"{key=} is not available") from exc
 
-    def __delitem__(self, key):
-        new_key = next((key_here for key_here in self if key.strip().lower() == key_here.lower()), key)
+    def __contains__(self, key: str) -> bool:  # type: ignore[override]
+        """To avoid cases sensitivity problems."""
+        return super().__contains__(key.lower().strip())
 
-        del self.data[new_key]
+    def __delitem__(self, key: str) -> None:
+        """To avoid cases sensitivity problems."""
+        super().__delitem__(key.lower().strip())
 
-    def diff(self, other):
-        """
-        Diff function for lobsterin. Compares two lobsterin and indicates which parameters are the same.
-        Similar to the diff in INCAR.
+    def diff(self, other: Self) -> dict[str, dict[str, Any]]:
+        """Compare two Lobsterin and find which parameters are the same.
+        Similar to the diff method of Incar.
 
         Args:
-            other (Lobsterin): Lobsterin object to compare to
+            other (Lobsterin): Lobsterin object to compare to.
 
         Returns:
-            dict with differences and similarities
+            dict: {"Same": same_params, "Different": diff_params}
         """
-        similar_param = {}
-        different_param = {}
-        key_list_others = [element.lower() for element in other]
+        same_params = {}
+        diff_params = {}
 
+        # Check self
         for k1, v1 in self.items():
-            k1_lower = k1.lower()
-            k1_in_other = next((key_here for key_here in other if key_here.lower() == k1_lower), k1_lower)
-            if k1_lower not in key_list_others:
-                different_param[k1.lower()] = {"lobsterin1": v1, "lobsterin2": None}
+            if k1 not in other:
```

Comment:
I am actually not sure this will work for all cases. Did you check with case sensitivity?

Reply:
I think it should, because we now store all keys as lower-case internally, so when we try to get a key from a `Lobsterin` instance, we don't need to cast it to lower case again anymore.

Yes I would have a look at the unit test and add one if there isn't any test on this.